apollo-router/src/configuration/mod.rs

```diff
@@ -85,21 +95,60 @@ pub struct Configuration {
 const APOLLO_PLUGIN_PREFIX: &str = "apollo.";
 const TELEMETRY_KEY: &str = "telemetry";
 
-fn default_listen() -> ListenAddr {
+fn default_graphql_listen() -> ListenAddr {
     SocketAddr::from_str("127.0.0.1:4000").unwrap().into()
 }
 
+// This isn't dead code! we use it in buildstructor's fake_new
+#[allow(dead_code)]
+fn test_listen() -> ListenAddr {
+    SocketAddr::from_str("127.0.0.1:0").unwrap().into()
+}
+
 #[buildstructor::buildstructor]
 impl Configuration {
     #[builder]
     pub(crate) fn new(
         server: Option<Server>,
+        graphql: Option<Graphql>,
+        health_check: Option<HealthCheck>,
+        sandbox: Option<Sandbox>,
         cors: Option<Cors>,
         plugins: Map<String, Value>,
         apollo_plugins: Map<String, Value>,
     ) -> Self {
         Self {
             server: server.unwrap_or_default(),
+            graphql: graphql.unwrap_or_default(),
+            health_check: health_check.unwrap_or_default(),
+            sandbox: sandbox.unwrap_or_default(),
+            cors: cors.unwrap_or_default(),
+            plugins: UserPlugins {
+                plugins: Some(plugins),
+            },
+            apollo_plugins: ApolloPlugins {
+                plugins: apollo_plugins,
+            },
+        }
+    }
+
+    // Used in tests
+    #[allow(dead_code)]
```

Comment:
NIT: If only used in tests, I feel like `#[cfg(test)]` is a better choice than allowing dead code.

Reply:
fixed in [8d9f60d](https://github.com/apollographql/router/pull/1718/commits/8d9f60d57547bd389b94ea5cc8632392bc9d7659)